src/extras/SceneUtils.js

```diff
@@ -37,9 +42,240 @@ var SceneUtils = {
 		scene.remove( child );
 		parent.add( child );
 
+	},
+
+	convertFromZUp: function ( root, animations ) {
+
+		// see https://gamedev.stackexchange.com/a/7932
+
+		// this matrix will change the coordinate system of an object hierarchy and
+		// animations (optional) from right-handed Z-up to right-handed Y-up
+
+		var matrix = new Matrix4().set(
+			1,   0,   0,   0,
+			0,   0,   1,   0,
+			0, - 1,   0,   0,
+			0,   0,   0,   1
+		);
+
+		var converter = new BasisConverter( matrix );
+		converter.convert( root, animations );
+
 	}
 
 };
 
+function BasisConverter( matrix ) {
+
+	this.matrix = matrix;
+
+}
+
+Object.assign( BasisConverter.prototype, {
+
+	convert: function ( root, animations ) {
+
+		var scope = this;
+
+		root.traverse( function( object ) {
+
+			// position, quaternion, scale
+
+			scope.convertMatrix4( object.matrix );
+			object.matrix.decompose( object.position, object.quaternion, object.scale );
+
+			// geometry
+
+			var geometry = object.geometry;
+
+			if ( geometry !== undefined &&  geometry.isBufferGeometry === true ) {
+
+				var position = geometry.attributes.position;
+				var normal = geometry.attributes.normal;
+
+				var matrix = scope.matrix;
+
+				if ( position !== null ) {
+
+					matrix.applyToBufferAttribute( position );
+
+				}
+
+				if ( normal !== null ) {
+
+					var normalMatrix = new Matrix3().setFromMatrix4( matrix );
+					normalMatrix.applyToBufferAttribute( normal );
+
+				}
+
+			}
+
+			// skinned mesh
+
+			if ( object.isSkinnedMesh ) {
+
+				scope.convertMatrix4( object.bindMatrix );
+				scope.convertMatrix4( object.bindMatrixInverse );
+
+				var boneInverses = object.skeleton.boneInverses;
+
+				for ( var i = 0, il = boneInverses.length; i < il; i ++ ) {
+
+					scope.convertMatrix4( boneInverses[ i ] );
+
+				}
+
+				// bones are already converted within the scene hierarchy
+
+			}
+
+		} );
+
+		// animations
+
+		if ( animations !== undefined ) {
+
+			for ( var i = 0, il = animations.length; i < il; i ++ ) {
+
+				var clip = animations[ i ];
+				var tracks = clip.tracks;
+
+				for ( var j = 0, jl = tracks.length; j < jl; j ++ ) {
+
+					this.convertKeyframeTrack( tracks[ j ] );
+
+				}
+
+			}
+
+		}
+
+	},
+
+	convertMatrix4: function() {
+
+		var xAxis = new Vector3();
+		var yAxis = new Vector3();
+		var zAxis = new Vector3();
+		var translation = new Vector3();
+
+		return function convertMatrix4( matrix ) {
```

Comment:
This is so cool!!! But i think i need some time to understand what kind of magic is going on there 😊 

> tinv is the transpose of t when t is orthonormal.

Okay, this step is clear to me! But the premultiplication and the subsequent multiplication with the inverse is something i don't comprehend yet. Give me some time for this one.